src/main/java/world/bentobox/chat/commands/island/IslandChatCommand.java

```diff
@@ -30,15 +30,26 @@ public void setup() {
 
     @Override
     public boolean canExecute(User user, String label, List<String> args) {
-        // TODO check rank
-        island = this.getIslands().getIsland(getWorld(), user);
-        return island != null;
+        boolean is = false;
+
+        if(is = this.getIslands().getIslandAt(user.getLocation()).isPresent())
+            island = this.getIslands().getIslandAt(user.getLocation()).get();
+        return is;
```

Comment:
Alternatively:

```
        island = this.getIslands().getIslandAt(user.getLocation()).orElse(null);
        return island != null;
```

Reply:
I tend to complicate things, since I'm new at coding, but I keep learning everyday more and more. Thanks! ^^